init.c

```diff
@@ -891,6 +891,19 @@ static int cutorch_setHeapTracking(lua_State *L)
   return 0;
 }
 
+static int cutorch_isManagedPtr(lua_State *L)
+{
+  THCState *state = cutorch_getstate(L);
+  if(lua_type(L, 1) != LUA_TNUMBER) {
+    THError("Must receive a ptr cast as a number");
+  }
+  void* ptr = (void* )luaL_optinteger(L, 1, 0);
+  struct cudaPointerAttributes attributes;
+  cudaPointerGetAttributes (&attributes, ptr);
```

Comment:
cudaPointerGetAttributes returns a `cudaError_t`, which should be checked. For CPU memory that's not managed or pinned it returns "cudaErrorInvalidValue", which should probably just be translated to `false` at the user level.